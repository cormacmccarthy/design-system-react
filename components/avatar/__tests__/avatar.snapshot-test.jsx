/* eslint-env jest */
import { testDOMandHTML } from '../../../tests/snapshot-helpers';

import SnapshotBase from '../__examples__/base';
import SnapshotEntityIcon from '../__examples__/entity-icon';
import SnapshotEntityInitials from '../__examples__/entity-initials';
import SnapshotUserIcon from '../__examples__/user-icon';
import SnapshotUserInitials from '../__examples__/user-initials';

import { AVATAR } from '../../../utilities/constants';

testDOMandHTML({
	name: 'Base',
	test,
	Component: SnapshotBase,
	ComponentKind: AVATAR
});

testDOMandHTML({
	name: 'Entity Icon',
	test,
	Component: SnapshotEntityIcon,
	ComponentKind: AVATAR
});

testDOMandHTML({
	name: 'Entity Initials',
	test,
	Component: SnapshotEntityInitials,
	ComponentKind: AVATAR
});

testDOMandHTML({
	name: 'User Icon',
	test,
	Component: SnapshotUserIcon,
	ComponentKind: AVATAR
});

testDOMandHTML({
	name: 'User Initials',
	test,
	Component: SnapshotUserInitials,
	ComponentKind: AVATAR
});
