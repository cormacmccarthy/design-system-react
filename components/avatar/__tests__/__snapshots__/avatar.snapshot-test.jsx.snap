// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Base DOM Snapshot 1`] = `
<div>
  <span
    className="slds-avatar slds-avatar_circle slds-avatar_medium"
  >
    <img
      alt="Person Name"
      onError={[Function]}
      src="https://lightningdesignsystem.com/assets/images/avatar2.jpg"
      title="user avatar"
    />
  </span>
</div>
`;

exports[`Base HTML Snapshot 1`] = `"<div><span class=\\"slds-avatar slds-avatar_circle slds-avatar_medium\\"><img alt=\\"Person Name\\" src=\\"https://lightningdesignsystem.com/assets/images/avatar2.jpg\\" title=\\"user avatar\\"/></span></div>"`;

exports[`Entity Icon DOM Snapshot 1`] = `
<div>
  <span
    className="slds-avatar slds-avatar_medium"
  >
    <span
      className="slds-icon_container slds-icon-standard-account"
      title={undefined}
    >
      <svg
        aria-hidden="true"
        className="slds-icon"
        style={undefined}
      >
        <use
          xlinkHref="/assets/icons/standard-sprite/svg/symbols.svg#account"
        />
      </svg>
      <span
        className="slds-assistive-text"
      >
        Account icon avatar
      </span>
    </span>
  </span>
</div>
`;

exports[`Entity Icon HTML Snapshot 1`] = `
"<div><span class=\\"slds-avatar slds-avatar_medium\\"><span class=\\"slds-icon_container slds-icon-standard-account\\"><svg aria-hidden=\\"true\\" class=\\"slds-icon\\"><use xlink:href=\\"/assets/icons/standard-sprite/svg/symbols.svg#account\\"></use></svg><span class=\\"slds-assistive-text\\">Account icon avatar</span></span>
  </span>
</div>"
`;

exports[`Entity Initials DOM Snapshot 1`] = `
<div>
  <span
    className="slds-avatar slds-avatar_medium"
  >
    <abbr
      className="slds-avatar__initials slds-icon-standard-account"
      title="Acme Communications"
    >
      AC
    </abbr>
  </span>
</div>
`;

exports[`Entity Initials HTML Snapshot 1`] = `"<div><span class=\\"slds-avatar slds-avatar_medium\\"><abbr class=\\"slds-avatar__initials slds-icon-standard-account\\" title=\\"Acme Communications\\">AC</abbr></span></div>"`;

exports[`User Icon DOM Snapshot 1`] = `
<div>
  <span
    className="slds-avatar slds-avatar_circle slds-avatar_medium"
  >
    <span
      className="slds-icon_container slds-icon-standard-user"
      title={undefined}
    >
      <svg
        aria-hidden="true"
        className="slds-icon"
        style={undefined}
      >
        <use
          xlinkHref="/assets/icons/standard-sprite/svg/symbols.svg#user"
        />
      </svg>
      <span
        className="slds-assistive-text"
      >
        User Icon Avatar
      </span>
    </span>
  </span>
</div>
`;

exports[`User Icon HTML Snapshot 1`] = `
"<div><span class=\\"slds-avatar slds-avatar_circle slds-avatar_medium\\"><span class=\\"slds-icon_container slds-icon-standard-user\\"><svg aria-hidden=\\"true\\" class=\\"slds-icon\\"><use xlink:href=\\"/assets/icons/standard-sprite/svg/symbols.svg#user\\"></use></svg><span class=\\"slds-assistive-text\\">User Icon Avatar</span></span>
  </span>
</div>"
`;

exports[`User Initials DOM Snapshot 1`] = `
<div>
  <span
    className="slds-avatar slds-avatar_circle slds-avatar_medium"
  >
    <abbr
      className="slds-avatar__initials slds-icon-standard-user"
      title="Annie Wilson"
    >
      AW
    </abbr>
  </span>
</div>
`;

exports[`User Initials HTML Snapshot 1`] = `"<div><span class=\\"slds-avatar slds-avatar_circle slds-avatar_medium\\"><abbr class=\\"slds-avatar__initials slds-icon-standard-user\\" title=\\"Annie Wilson\\">AW</abbr></span></div>"`;
